src/amber/router/pipe/session.cr

```diff
@@ -63,8 +65,12 @@ module Amber
       end
 
       class Params < Hash(String, String)
+        property changed = false
+
         def []=(key : String | Symbol, value : V)
```

Comment:
Is there other ways of adding or changing values in a hash?  What about `merge!`?